### ISSUE TITLE
[css-text][text-autospace] Is halfwidth Kana "non-ideographic letters"?

### ISSUE THREAD

**Issue by kojiishi** [w3c/csswg-drafts#9471]
From the [8.5.2. Text Spacing Character Classes](https://drafts.csswg.org/css-text-4/#text-spacing-classes), halfwidth Kana belongs to "non-ideographic letters". This means that there will be an auto-space between ideographs and halfwidth Kana. This seems like an overlook.

The question is whether they should be "ideograph" or not. Terminology-wise, Kana is derived from Han, so it's closer to ideograph than to non-ideograph. Behavior-wise, I don't think authors expect auto-space between ideograph/fullwidth Kana and halfwidth Kana.

So I think we should classify them not to belong to either "ideographs" nor "non-ideographic letters". I'll check with JLTF about this separately.

**Comment by kojiishi** [comment]
[Japanese discussion here](https://lists.w3.org/Archives/Public/public-i18n-japanese/2023OctDec/0160.html)

**Comment by kojiishi** [comment]
3 opinions in the Japanese discussion, all agreed:
* No auto-space between halfwidth Kana and "ideograph".
* No auto-space between halfwidth Kana and "non-ideographic letters/numerals".

**Comment by frivoal** [comment]
@kojiishi  https://github.com/w3c/csswg-drafts/pull/9503 tries to address this (and a some other things), can you have a look?

**Comment by nt1m** [comment]
cc @fantasai @vitorroriz for WebKit input.

**Comment by frivoal** [comment]
As pointed out in https://github.com/w3c/csswg-drafts/issues/9501#issuecomment-1779704918, we might have a related question about Bopomofo. By the way, how about Hangul? Maybe it should be treated similarly to Bopomofo?
Should these be made to fit in the `ideographs` category, or `non-ideographic letters`, or neither like half-width katakana?

What about other east asian scripts, such as:
* https://util.unicode.org/UnicodeJsps/list-unicodeset.jsp?a=[:Script_Extensions=Khitan_Small_Script:]
* https://util.unicode.org/UnicodeJsps/list-unicodeset.jsp?a=[:Script=Tangut:]
* https://util.unicode.org/UnicodeJsps/list-unicodeset.jsp?a=[:Script_Extensions=Nushu:]

All these have East_Asian_Width set to 'Wide', but not 'FullWidth'

**Comment by kojiishi** [comment]
> how about Hangul?

@jungshik Please see the comment above. Would Korean readers want 1/8em auto-spacing between Hangul and Han Ideograph?

> What about other east asian scripts, such as:

I would prefer not to go too far. This isn't a logical choice but just a pattern used for a long time, so the right answer doesn't exist, and updating the set from future feedback is likely possible.

**Comment by fantasai** [comment]
@frivoal Khitan, Nushu, and Tangut should be categorized as ideographic. In fact, all Wide characters should be categorized this way. I don't think users would expect spacing between them and Han ideographs.

**Comment by xfq** [comment]
I agree with Khitan and Tangut (and maybe Jurchen and Classical Yi).

However, Nüshu is vastly different from Han characters, and its ideal character frame is rectangular, with the width of the character less than the height of the character. We will discuss this in the clreq group.

**Comment by r12a** [comment]
I'm no expert on Khitan Small Script, but i think it is a bit special since characters are arranged in 2-dimensional groups, separated by spaces.  Here's a sample:

<img width="1113" alt="Screenshot 2023-11-14 at 11 57 25" src="https://github.com/w3c/csswg-drafts/assets/4839211/cb1e8965-32ee-49f5-b8f4-99455c6e85b7">

https://r12a.github.io/scripts/samples/index.html?script=kits#result

**Comment by xfq** [comment]
The Khitan small script sequences do have spaces between them, and the spaces are there to terminate the words.

The problem with the Noto font, though, is that it deviates from the aesthetic and design of the script. The width of the top part and the bottom part should be the same. The following screenshot is a correct example:

![KSS_Vertical_Word](https://github.com/w3c/csswg-drafts/assets/2863444/33d4964e-1c1d-4cab-8f5b-652cf7b17982)

If there is a zero width space or no space between between runs of text in Khitan small script and non-ideographic letters/numerals, the layout engine should add the extra spacing. If there is already a space (U+0020) between the text in Khitan small script and non-ideographic letters/numerals, there is no need for the layout engine to add extra spacing.

**Comment by xfq** [comment]
FWIW, there is also a proposal in Unicode about this: https://www.unicode.org/L2/L2023/23283-auto-spacing-prop.pdf

**Comment by kojiishi** [comment]
This is resolved by adopting the UTR#59 in #11013.
